src/goto-instrument/cover.h

```diff
@@ -66,6 +70,13 @@ void instrument_cover_goals(
 
 void parse_cover_options(const cmdlinet &, optionst &);
 
+bool instrument_cover_goals(
```

Comment:
This one definitely shouldn't taken an optional, since there is already an overload that doesn't require one at all - how would I pick between calling with with `{}` and without the argument entirely?

Reply:
The reason I left the other one there was backwards compatibility of public methods. I'm happy to go with either solution, I can change all known usages of the not-even-optional overload to using optional. Up to you @thk123 and whoever else reviews this PR.

Reply:
Optional is not required for backwards compatibility, as there already exists an overload that doesn't take the parameter. Happy to talk through this with you, but this one should really be addressed IMO.